src/main/java/org/folio/innreach/domain/service/impl/MaterialTypeMappingServiceImpl.java

```diff
@@ -0,0 +1,115 @@
+package org.folio.innreach.domain.service.impl;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.UUID;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import org.folio.innreach.domain.entity.CentralServer;
+import org.folio.innreach.domain.entity.MaterialTypeMapping;
+import org.folio.innreach.domain.exception.EntityNotFoundException;
+import org.folio.innreach.domain.service.MaterialTypeMappingService;
+import org.folio.innreach.dto.MaterialTypeMappingDTO;
+import org.folio.innreach.dto.MaterialTypeMappingsDTO;
+import org.folio.innreach.mapper.MaterialTypeMappingMapper;
+import org.folio.innreach.repository.MaterialTypeMappingRepository;
+
+@RequiredArgsConstructor
+@Service
+@Transactional
+public class MaterialTypeMappingServiceImpl implements MaterialTypeMappingService {
+
+  private final MaterialTypeMappingRepository repository;
+  private final MaterialTypeMappingMapper mapper;
+
+  
+  @Override
+  @Transactional(readOnly = true)
+  public MaterialTypeMappingsDTO getAllMappings(UUID centralServerId, Integer offset, Integer limit) {
+    var example = mappingExampleWithServerId(centralServerId);
+
+    var totalCount = repository.count(example);
```

Comment:
Do we really need this additional count query?

Reply:
Yes, we do.. It calculates the total number of records not limited by `limit` and `offset` parameters. It's a common agreement in Folio to return the total count for such kind of queries. 
I know it's settings and in most cases query will be retrieve everything, but still

Reply:
You don't need to query for it explicitly.
Page object can return the total number of elements in the database, 
Please, check **mappings.getTotalElements()**

Reply:
Thanks for the tip. Will take a look at the method